### ISSUE TITLE
Feat: Still look into how a custom list can be supported

### ISSUE THREAD

**Issue by RedBrumbler** [RedBrumbler/Quest-BSML#64]
Possible ideas:

 - forward the ITableData calls to a BSMLValue that the modder provides from the host object
 - Let the modder give a fully specified type (aka namespaze="BSML" class="CoolTableData" OR type="BSML::CoolTableData") and require it to be a MonoBehaviour and implement the ITableDataSource interface, and let users make a custom list that way through BSML
 - Emulate the PC experience with a BSMLValue that's a list of objects which get a Setup / Reuse method called after the cells get made. This is hard for reusing cells though so very unlikely

Bottom line is that not reusing cells is unacceptable, so if we can't do that, we're not going to be doing this
